### PR TITLE
Document status enums.

### LINKED ISSUE #573: Docs: document status enums
https://github.com/JuliaOpt/MathOptInterface.jl/blob/81c115594f6cc0dae3d288d69ffa428eadc0f5b7/src/attributes.jl#L726-L742

https://github.com/JuliaOpt/MathOptInterface.jl/blob/81c115594f6cc0dae3d288d69ffa428eadc0f5b7/src/attributes.jl#L710

https://github.com/JuliaOpt/MathOptInterface.jl/blob/81c115594f6cc0dae3d288d69ffa428eadc0f5b7/src/attributes.jl#L719

### PR DESCRIPTION
**Pull request by odow**
Closes #573

### CODE REVIEW

File: src/attributes.jl [hunk 1]
```diff
@@ -729,18 +770,27 @@ To be documented: `NumericalError`, `InvalidModel`, `InvalidOption`, `Interrupte
...
+* `NoSolution`: the result vector is empty.
+* `FeasiblePoint`: the result vector is a feasible point.
+* `NearlyFeasiblePoint`: the result vector is feasible if some constraint
+  tolerances are relaxed.
+* `InfeasiblePoint`: the result vector is an infeasible point.
+* `InfeasibilityCertificate`: the result vector is an infeasibility certificate.
+  Given a dual feasible solution, an infeasibility certificate in the primal
+  implies that the dual is unbounded, given a primal feasible solution, an
+  infeasibility certificate in the dual implies that the primal is unbounded.
 * `NearlyInfeasibilityCertificate`
```
Comment (odow) [inline src/attributes.jl]: @mlubin how should this be interpreted?

File: src/attributes.jl [hunk 2]
```diff
@@ -729,18 +770,27 @@ To be documented: `NumericalError`, `InvalidModel`, `InvalidOption`, `Interrupte
...
 * `ReductionCertificate`
```
Comment (odow) [inline src/attributes.jl]: Ditto

File: src/attributes.jl [hunk 3]
```diff
@@ -729,18 +770,27 @@ To be documented: `NumericalError`, `InvalidModel`, `InvalidOption`, `Interrupte
...
 * `NearlyReductionCertificate`
```
Comment (odow) [inline src/attributes.jl]: Ditto

File: src/attributes.jl [hunk 4]
```diff
@@ -721,14 +721,14 @@ The optimizer stopped because of some user-defined limit.
...
+  solutions. This is often used in MIPs to get the solver to return the first
   feasible solution it encounters.
 * `MemoryLimit`: the algorithm stopped because it ran out of memory.
 * `ObjectiveLimit`: the algorthm stopped because it found a solution better than
   a minimum limit set by the user.
 * `NormLimit`: the algorithm stopped because of a criteria involving a norm.
-  This is typically used to terminate an algorithm once successive iterations
-  result in a solution that is within some distance of the previous solution.
+  This is typically used to indicate that an algorithm stopped after
+  encountering a sequence of diverging iterations.
```
Comment (mlubin) [inline src/attributes.jl]: Diverging sequence is maybe too specific. The limit means that the norm of an iterate became too large.

File: src/attributes.jl [hunk 5]
```diff
@@ -779,9 +779,8 @@ The values indicate how to interpret the result vector.
   Given a dual feasible solution, an infeasibility certificate in the primal
   implies that the dual is unbounded, given a primal feasible solution, an
   infeasibility certificate in the dual implies that the primal is unbounded.
-* `NearlyInfeasibilityCertificate`
-* `ReductionCertificate`
```
Comment (mlubin) [inline src/attributes.jl]: Remove these from the enum also

File: src/attributes.jl [hunk 6]
```diff
@@ -692,34 +692,75 @@ struct TerminationStatus <: AbstractModelAttribute end
...
     TerminationStatusCode
 
-An Enum of possible values for the `TerminationStatus` attribute.
-This attribute is meant to explain the reason why the optimizer stopped executing.
+An Enum of possible values for the `TerminationStatus` attribute. This attribute
+is meant to explain the reason why the optimizer stopped executing.
 
 ## OK
 
 These are generally OK statuses.
```
[the remark this answers is not in the document]
Comment (IainNZ) [reply src/attributes.jl]: `"sweet as" statuses`

**Comment by odow** [comment]
Is this better?

**Comment by mlubin** [comment]
> Is this better?

LGTM

**Pull request merged**